src/reference/asciidoc/mongodb.adoc

```diff
@@ -290,3 +290,116 @@ and other attributes that are common across all other inbound adapters (e.g., 'c
 The example above is relatively simple and static since it has a literal value for the `collection-name`.
 Sometimes you may need to change this value at runtime based on some condition.
 To do that, simply use `collection-name-expression` where the provided expression can be any valid SpEL expression.
+
+=== MongoDB Outbound Gateway
+
+The MongoDb Outbound Gateway allows you to query a database by sending a Message to its request channel.
+The gateway will then send the response to the reply channel.
+The Message payload and headers can be used to specify the query.
+
+[source,xml]
+----
+<int-mongodb:outbound-gateway id="gatewayQuery"
+    mongodb-factory="mongoDbFactory"
+    mongo-converter="mongoConverter"
+    query="{firstName: 'Bob'}"
+    collection-name="foo"
+    request-channel="in"
+    reply-channel="out"
+    entity-class="org.springframework.integration.mongodb.test.entity.Person"/>
+----
+
+* `collection-name` or `collection-name-expression` - Identifies the name of the MongoDb collection to use.
+* `mongo-converter` - reference to an instance of `o.s.data.mongodb.core.convert.MongoConverter` to assist with converting a raw java object to a JSON document representation
+* `mongodb-factory` - reference to an instance of `o.s.data.mongodb.MongoDbFactory`
+* `mongo-template` - reference to an instance of `o.s.data.mongodb.core.MongoTemplate` (NOTE: you can not have both mongo-template and mongodb-factory set)
+* `entity-class` - The fully qualified name of the entity class to to be passed to find(..) or findOne(..) method in MongoTemplate. If this attribute is not provided the default value is com.mongodb.DBObject
```

Comment:
Double `to`. 

`find(..)` and so should be wrapped to code style.
And the same for the `DBObject` although we decided with you that `Document` should be there now.

Plus take a look into other chapters (e.g. `amqp.adoc`) how call-outs are done - to have cross-links from the code snippet to the description and back.